src/main/java/net/minecraftforge/client/ArmorOverlayHandler.java

```diff
@@ -0,0 +1,96 @@
+package net.minecraftforge.client;
+
+import net.minecraft.client.model.ModelBase;
+import net.minecraft.client.renderer.GlStateManager;
+import net.minecraft.client.renderer.texture.TextureManager;
+import net.minecraft.entity.EntityLivingBase;
+import net.minecraft.inventory.EntityEquipmentSlot;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+import net.minecraftforge.common.ForgeModContainer;
+
+/**
+ * Offshore handler for the Armors' extra passes. Specifically for reducing patch size & maintaining readability.
+ * */
+public class ArmorOverlayHandler
+{
+
+    private static final java.util.Map<net.minecraftforge.fml.common.registry.RegistryDelegate<Item>, IArmorOverlay> armorOverlayMap = com.google.common.collect.Maps.newHashMap();
+
+    /**
+     * Number of passes to be run.
+     * */
+    private static final byte NUM_PASS = 2;
+
+    private ArmorOverlayHandler(){}
+
+    /**
+     * Handles the execution of overlays.
+     * */
+    public static boolean run(TextureManager textureManager, ItemStack stack, EntityLivingBase wearer, float partialTicks, EntityEquipmentSlot slot, ModelBase model, float swing, float swingScale, float headYaw, float headPitch, float scale)
+    {
+        if(ForgeModContainer.forgeArmorOverlay)
+        {
+            IArmorOverlay current = armorOverlayMap.get(stack.getItem().delegate);
+            float[] vec;
+            int color;
+            float time = wearer.ticksExisted + partialTicks;
+            if(current == null)
+                current = IArmorOverlay.VANILLA;
```

Comment:
More or less. The actual process has had a bunch of checks ripped out of it and the vertex upload nested into the OverlayHandler so it runs with a bit less overhead than the vanilla method. The return is meant purely as a configuration thing, as otherwise it'd always return false to just use handler methods. As @mezz pointed out config support is probably moot point so I'll probably just phase it out.